Upgrade perseus to npm published version.

rtibbles opened the pull request:
## Summary
* Upgrades our perseus version to the most recently published version on NPM after Khan Academy re-open sourced the code base.
* Adds required dependency components `KatexProvider` and `TeX` for handling rendering, borrowing the latter from the the deprecated [Khan React components](https://github.com/Khan/react-components/blob/fee9ceb0ff7dfe40370465101dd86f1def33a9d5/js/tex.jsx)
* Updates the build scripts to simplify to just vendoring the Perseus JS and CSS for small tweaks and monkey patches
* Moves some of the monkey patching into runtime code to simplify this and make more apparent
* Uses specific imports from React and React-DOM to be clear about which elements we are using
* Updates translation extraction to pull from the typescript files that Perseus packages have that expose the strings
* Uses updated React context injection to provide strings for Perseus.
* Removes the standalone webpack config for prebuilding the Perseus dist, and removes all dependencies related to that
* Removes all previously patched globals that Perseus no longer relies on

## Reviewer guidance
Test exercises in lots of different languages, and with lots of different exercise types! The most problematic are probably the more complex mathematical ones - graphs, lots of mathematical formulae etc.

----

Fixes #8362 

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

pcenov commented:
Hi @rtibbles, I was able to identify just the following issues by extensively testing various perseus exercises in multiple languages (valid in quiz, lesson or library context):

| Issue| Screenshot |
:-------------------------:|:-------------------------:
| Multiple choice questions - It's not possible to deselect a selected answer | ![2022-10-06_18-03-43](https://user-images.githubusercontent.com/79847249/194350168-838432b3-7f30-4e12-b01c-c284e8f00311.png)|
:-------------------------:|:-------------------------:
| Fields with option to enter exponents or fractions are not keeping the selected value | ![2022-10-06_17-00-57](https://user-images.githubusercontent.com/79847249/194350949-ba54c5af-0244-49cf-a8a2-42bc77d7b28c.png)|
:-------------------------:|:-------------------------:
| Multiple "Not allowed to use local resource" errors in the console with seemingly no user facing consequences |  ![2022-10-06_18-22-50](https://user-images.githubusercontent.com/79847249/194356793-d8d5a441-ff8c-43db-a633-0b9eb67f62e5.png)|
:-------------------------:|:-------------------------:

The issues above can be replicated by importing Khan Academy (English - US curriculum) > Math by grade > 8th Grade > Illustrative Mathematics

rtibbles commented:
Thanks @pcenov - I assumed that there is no way this could have gone as smoothly as I thought it had! Will look into these issues, thanks for the precise reference for replication.

nucleogenesis commented on kolibri/plugins/perseus_viewer/assets/src/views/PerseusRendererIndex.vue:
```diff
@@ -88,41 +88,91 @@
+  // so that even if we have multiple instances of the PerseusRenderer
+  // instantiated at once, we can still render images from all of them.
+  // This also allows us to only monkey patch the Util functions once as
+  // well, and prevent duelling components from overriding each other.
```
Multi-modal friendliness <3

nucleogenesis reviewed (dismissed):
Code LGTM so long as it passes manual QA. Added comments for context are much appreciated (probably mostly by you some years later when you find the need to revendor Perseus).

pcenov commented:
Hi @rtibbles, after testing the latest build the issue with the exponents and fractions disappearing is still valid while the one with the single/multiple choice questions is partially fixed as it is now possible to deselect a selected answer but there are some blue lines displayed each time I make a selection, a yellow border displayed if I deselect a selected answer and also if I navigate to another question after having deselected my answer when I go back to the same question the answer is selected again. Videos:

| Issue| Screenshot |
:-------------------------:|:-------------------------:
| Single/Multiple choice questions issues | https://user-images.githubusercontent.com/79847249/210579142-8453ea15-e769-443d-8953-cbdf8c3d0f5b.mp4 |
:-------------------------:|:-------------------------:
| Fields with option to enter exponents or fractions are not keeping the selected value | https://user-images.githubusercontent.com/79847249/210579236-32854c6d-8582-4e02-a991-a837b73d434a.mp4 |
:-------------------------:|:-------------------------:

pcenov commented:
@rtibbles following a discussion in Slack I tested the reported issue with the missing bar graph and I can confirm that I am able to replicate it both at https://content-demo.learningequality.org/en/learn/#/topics/t/b18a118b93dc52fe9f1c68f9e418aaf5/search and when using the latest build from this PR.
The difference is that at content-demo the bar graph is not being displayed at all while it is partially displayed in the latest build:

![2023-01-06_17-26-28](https://user-images.githubusercontent.com/79847249/211048168-3eb9fe11-14f7-49f3-a87e-17d19a75e456.png)

Here's also a video of what is being displayed in the latest build:

https://user-images.githubusercontent.com/79847249/211048284-aeca5ce2-6ca1-40fa-af8a-3b0afddba8bc.mp4

I also noticed an issue with one multiple-select question where the text for the number of answers that need to be selected is not displayed correctly:

![2023-01-06_18-00-47](https://user-images.githubusercontent.com/79847249/211050682-26f64f97-27eb-4f05-9dc9-d6c97b161ee3.png)

rtibbles commented:
@pcenov I have looked at the bar graph issue, and determined that it is a problem with the cheffed question, not the renderer, so will need to be fixed during import.

I have made updates to much later versions of Perseus, but I do not know if the reported issues will be fixed by that, so I'll have to do more testing to look into the other issues you've identified here.

pcenov commented:
@rtibbles, good to see progress made on that one! Should I test it again to see what's the current state of affairs?

rtibbles commented:
That would be great - but definitely much lower priority than anything else on your plate right now! This would not be released until 0.17.0 at the earliest.

rtibbles committed:
Upgrade perseus and fix bugs in expression responses.

rtibbles committed:
Allow binary expressions as message definitions.
Handle errors slightly more gracefully.

rtibbles committed:
Handle enter key to give answer.

pcenov commented:
Hi @rtibbles - I attempted to retest this using the latest build assets but there are 2 main blockers:

1. After I answer a question and then return back to it I can see that none of the answers are saved:

https://github.com/learningequality/kolibri/assets/79847249/67fc04e8-98ea-4e17-872f-8e7e64c21084

2. All the tables are not displayed:

![2024-05-28_11-35-29](https://github.com/learningequality/kolibri/assets/79847249/da0dc05c-32f4-4791-935e-7ce7f0e527f7)

![2024-05-28_11-38-02](https://github.com/learningequality/kolibri/assets/79847249/b4894e50-8100-4a8f-b3fb-af934bddec0a)

![2024-05-28_11-39-49](https://github.com/learningequality/kolibri/assets/79847249/5d658e1e-72e9-42df-914e-2c2d243398d3)

Here's a video as well:

https://github.com/learningequality/kolibri/assets/79847249/5d6f151f-b285-46d3-9d62-3efd2311e0c5

I'm using exercises from `Khan Academy (English - CBSE India Curriculum) > Math by grade (India)`

rtibbles commented:
Thanks @pcenov - looks like there is an issue with restoring the saved answers, and also with one particular string (judging from the errors in the console).

rtibbles committed:
Group dependabot updates for Khan Academy namespaced packages.

rtibbles commented:
@pcenov I managed to reproduce the answer updating errors, and I seem to have resolved them in my testing.

The string errors, I did not directly reproduce, but I deduced they were caused by differences in how translations are handled in the development environment versus production - I have updated the translation handling to do something slightly different so it should behave the same way for both now.

pcenov commented:
Thanks @rtibbles - I noticed just the following few issues so this one is almost ready:

1. On a quiz created with questions from the QA Channel (Identifying Rectangles, Squares, Circles, and Triangles Practice) I cannot edit the given answers as when I go back the entire question has disappeared:

https://github.com/learningequality/kolibri/assets/79847249/c8daba59-89ba-46cb-aec9-6603d9206f0d

2. For some exercises such as `Khan Academy (English - CBSE India Curriculum) - Math by grade (India) - Class 8 - Algebraic expressions and identities` I see an exclamation mark and the following error in the console: `TypeError: strings.labelValue is not a function at Object.labelValue`:

https://github.com/learningequality/kolibri/assets/79847249/2b1a7f9d-eb29-45b2-8689-7f19d9740d59

Observed the same in in `Khan Academy (English - US curriculum) Math - Get ready for Geometry - Get ready for analytic geometry - Horizontal & vertical lines` as well:

![2024-06-05_18-27-34](https://github.com/learningequality/kolibri/assets/79847249/0374b7bb-b020-49a2-a566-03a39f7664db)


3. In `Khan Academy (English - CBSE India Curriculum) - Math by grade (India) - Class 5 - Smart charts Picture graphs` the icon and the text are not positioned on the same line:

![2024-06-05_18-00-19](https://github.com/learningequality/kolibri/assets/79847249/c6fa162f-7f48-46a7-92a6-aa032816cb63)

rtibbles commented:
Thank you @pcenov - the detailed text of the console errors is especially helpful here!

rtibbles committed:
Move perseusFile handling into module scope to allow efficient and
non-colliding recurrence of perseus rendering within the same page.

rtibbles commented:
Thanks to your sleuthing, @pcenov I discovered that a whole chunk of strings were not being properly extracted, so I think I have now vanquished the string issues here.

In my ongoing testing I discovered some more foibles with the state restoration, especially with quizzes made from the QA Channel (Identifying Rectangles, Squares, Circles, and Triangles Practice) exercise you highlighted. I think I have resolved all of these now though.

I discovered that the bug with the display of the image inline with text was actually a bug in Perseus itself - this has been reported to Khan Academy, but I have also added a workaround in our code as well, so hopefully it should be resolved.

Lastly, I did some testing in the quiz creation workflow to make sure that previewing multiple different questions at once was working as intended, it seemed to be working robustly, including when questions are removed (and so their renderers are cleaned up and state removed), I have made some updates to try to make sure this is as robust and efficient as possible, so some testing of those workflows would be appreciated!

pcenov commented:
Hi @rtibbles - happy to let you know that this time I didn't find any new issues with the rendering of the perseus exercises however when I create a quiz the first section with questions always gets duplicated and also my answers keep disappearing:

https://github.com/learningequality/kolibri/assets/79847249/54e94ce3-4a54-40ab-8993-97b3b0f7241b

Here's the error in the console:

```
PerseusRendererIndex.vue:386 TypeError: Cannot read properties of undefined (reading 'imageUrls')
    at replaceImageUrls (PerseusRendererIndex.vue:151:1)
    at Sub.restoreSerializedState (PerseusRendererIndex.vue:497:1)
    at Sub.setAnswer (PerseusRendererIndex.vue:519:1)
    at Sub._resetState (PerseusRendererIndex.vue:433:1)
    at Sub.<anonymous> (PerseusRendererIndex.vue:443:1)
    at Sub.on (vue.runtime.esm.js:3840:10)
    at invokeWithErrorHandling (vue.runtime.esm.js:1863:26)
    at Vue.$emit (vue.runtime.esm.js:3903:9)
    at ref (PerseusRendererIndex.vue:386:1)
    at Sj (react-dom.production.min.js:255:383)
```

rtibbles commented:
Great, thanks @pcenov, that seems directly related to my most recent changes so I will chase this down!

rtibbles commented:
Hi @pcenov - the duplicate rendering seems to be unrelated to this PR, and should be resolved by #12182. I have prevented the error from occurring, and it should properly display, except that items in section 2 will always actually be the item from section 1 until #12182 is merged.

pcenov commented:
Hi @rtibbles I confirm that the error is fixed. While regression testing the Android app I noticed that it's no longer possible to input the answer to an exercise as the input options are not being displayed. Let me know if I should file that as a separate issue.
Logs:  [logs.zip](https://github.com/user-attachments/files/15787021/logs.zip)

rtibbles commented:
Hrm, this is almost certainly an issue with this PR, so I'll fix it here - thanks for your diligence!

rtibbles committed:
Update to perseus 11.5.1

rtibbles committed:
Add the math-input styling into dist, as it is not importable from the package.

rtibbles committed:
Instantiate the MobileKeypad components to ensure it is available for mobile input.
Update styling through the assessment stack to ensure we can position it properly.

rtibbles commented:
Hi @pcenov - I think that the keypad should now be reinstated, and hopefully working as intended (although I do notice there is a visual difference to it from before).

![Screenshot from 2024-06-16 18-56-06](https://github.com/learningequality/kolibri/assets/1680573/856e883e-26a8-4acf-9a5e-f06e14c59c7c)

pcenov commented:
Hi @rtibbles - I confirm that the keyboard is displayed now and it works correctly when viewing an exercise from the Library page. However when you create a quiz and attempt to fill in an answer the keyboard is displayed too high and also doesn't fit the screen properly (at least on my Note 10 Samsung phone which has a very large screen) so that the Back and Close button are invisible to the user unless the user drags the keyboard to the left. See the video:

https://github.com/learningequality/kolibri/assets/79847249/4f89f70b-6a37-4a73-8f4f-57af20481006

rtibbles commented:
Hi @pcenov I think I came up with a solution that seems to work well for both the exercise and quiz contexts, hopefully this is robust!

![image](https://github.com/learningequality/kolibri/assets/1680573/b6989507-347c-46d4-9510-2c3cbf01cfd1)

pcenov commented:
Hi @rtibbles I confirm that now the keyboard fits better withing the screen in the Android app. Here's how it looks when compared to the previous implementation - might be worth investigating whether it can be further improved in a follow-up PR:

![2024-06-18_14-32-58](https://github.com/learningequality/kolibri/assets/79847249/35dbaeb5-a300-4d96-a54b-a52e735f02b9)

Also when testing the .deb build if I shrink the browser width to less than 820px the on-screen keyboard pops up, which shouldn't be so, right:

![2024-06-18_14-53-50](https://github.com/learningequality/kolibri/assets/79847249/de2e08e9-3529-45a8-afb4-e99bfe8a4c75)

rtibbles commented:
@pcenov this seems to have been a change by Khan Academy within perseus to display the keypad even on desktop - I think I'm ok with leaving it in, as it still allows use of the regular keys to enter the response. I think this may be a replacement for some of the previous tooltips they used to suggest permissable values (like pi, etc).

rtibbles commented:
Oh.. or maybe not, as although it pops up, it doesn't actually allow input via the keypad.

rtibbles committed:
Always use custom keypad, regardless of input modality.

rtibbles commented:
OK, I have updated it so that the keypad does always show, regardless of modality, due to the lack of tooltips to offer prompts for how to answer, but in doing so, have also made sure that it works regardless of modality, and can still do keyboard input as well.

marcellamaki commented on kolibri/plugins/perseus_viewer/assets/dist/index.css:
```diff
@@ -0,0 +1,4547 @@
+    line-height: 1.2;
+}
+
+table.periodic-table td.element div.atomic-num {
```
Is this intentionally empty? (it looks like one of the few places there were more recent modifications which is why I ask)

rtibbles replied:
Ah - the dist files are just directly copied from KA's npm packages, so :shrug: - I take no responsibility for these, except the minor manipulations we make to have valid font file URLs in them.

marcellamaki reviewed (commented):
minimal feedback so far. I think reading commit by commit, the code is.... as comprehensible as it can be (most of the complexity related to our use of perseus in general, not the method of implementation)

rtibbles committed:
Upgrade perseus to npm published version.

rtibbles committed:
Update perseus to 22.7.0.
Upgrade to ServerItemRenderer API.
Update message parsing and extraction.

pcenov commented:
Hi @rtibbles,
I confirm that the onscreen keyboard is now always displayed in both full-screen of a desktop browser and in mobile view. It works correctly in the Android app, but it's best to further discuss what can be improved there from usability point of view as for some exercises the keyboard is pushed at the lower end of the screen and the user has to be smart enough to scroll up in order to use it. :)

Also spotted  the following issues:

1. I was having trouble with installing the latest build assets, getting the following error when I run Kolibri:

Ubuntu:

```
Running migrations:
  Applying morango.0003_auto_20170519_0543...Error: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 189, in invoke
    initialize(**get_initialize_params())
  File "/usr/lib/python3/dist-packages/kolibri/utils/main.py", line 342, in initialize
    update(version, kolibri.__version__)
  File "/usr/lib/python3/dist-packages/kolibri/utils/main.py", line 396, in update
    _migrate_databases()
  File "/usr/lib/python3/dist-packages/kolibri/utils/main.py", line 109, in _migrate_databases
    call_command("migrate", interactive=False, database=database)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 181, in call_command
    return command.execute(*args, **defaults)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/commands/migrate.py", line 246, in handle
    fake_initial=fake_initial,
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/migrations/operations/fields.py", line 353, in database_forwards
    to_model._meta.get_field(self.new_name),
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/sqlite3/schema.py", line 115, in alter_field
    ) % (model._meta.db_table, old_field_name))
django.db.utils.NotSupportedError: Renaming the 'morango_scopedefinition'.'scope_id' column while in a transaction is not supported on SQLite < 3.26 because it would break referential integrity. Try adding `atomic = False` to the Migration class.
```
Windows:

```
C:\Users\Kolibri>kolibri start --foreground
←[37mINFO     2024-06-19 17:13:58,310 Option DEBUG in section [Server] being overridden by environment variable KOLIBRI_DEBUG←[0m
←[37mINFO     2024-06-19 17:13:58,313 Option DEBUG_LOG_DATABASE in section [Server] being overridden by environment variable KOLIBRI_DEBUG_LOG_DATABASE←[0m
←[37mINFO     2024-06-19 17:13:58,313 Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE←[0m
←[37mINFO     2024-06-19 17:13:58,313 Attempting to setup using pre-migrated databases←[0m
c:\python36\lib\site-packages\kolibri\dist\morango\models\fields\crypto.py:23: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
  from cryptography.hazmat.backends import default_backend
←[37mINFO     2024-06-19 17:14:01,438 Running Kolibri with the following settings: kolibri.deployment.default.settings.base←[0m
←[37mINFO     2024-06-19 17:14:01,750 New install, version: 0.17.0a0.dev0+git.43.gf4569e12←[0m
←[37mINFO     2024-06-19 17:14:01,750 Running update routines for new version...←[0m
Operations to perform:
  Apply all migrations: analytics, auth, bookmarks, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, sessions
Running migrations:
  Applying morango.0003_auto_20170519_0543...Error: Traceback (most recent call last):
  File "c:\python36\lib\site-packages\kolibri\utils\cli.py", line 189, in invoke
    initialize(**get_initialize_params())
  File "c:\python36\lib\site-packages\kolibri\utils\main.py", line 342, in initialize
    update(version, kolibri.__version__)
  File "c:\python36\lib\site-packages\kolibri\utils\main.py", line 396, in update
    _migrate_databases()
  File "c:\python36\lib\site-packages\kolibri\utils\main.py", line 109, in _migrate_databases
    call_command("migrate", interactive=False, database=database)
  File "c:\python36\lib\site-packages\kolibri\dist\django\core\management\__init__.py", line 181, in call_command
    return command.execute(*args, **defaults)
  File "c:\python36\lib\site-packages\kolibri\dist\django\core\management\base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "c:\python36\lib\site-packages\kolibri\dist\django\core\management\base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "c:\python36\lib\site-packages\kolibri\dist\django\core\management\commands\migrate.py", line 246, in handle
    fake_initial=fake_initial,
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\migrations\executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\migrations\executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\migrations\executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\migrations\migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\migrations\operations\fields.py", line 353, in database_forwards
    to_model._meta.get_field(self.new_name),
  File "c:\python36\lib\site-packages\kolibri\dist\django\db\backends\sqlite3\schema.py", line 115, in alter_field
    ) % (model._meta.db_table, old_field_name))
django.db.utils.NotSupportedError: Renaming the 'morango_scopedefinition'.'scope_id' column while in a transaction is not supported on SQLite < 3.26 because it would break referential integrity. Try adding `atomic = False` to the Migration class.
```
2. I did manage to install the build in another Ubuntu VM and the issues I am seeing with the on-screen keyboard are:
- when answering the questions in a quiz it seems that the focus is not in the input field so that the selected value from the keyboard is not entered in the field. But even if I manually input it sometimes it just disappears when I revisit the question:

https://github.com/learningequality/kolibri/assets/79847249/55dc7acf-d412-4bfd-accf-34de915cf470

Here's the error in the console:
[browser-console-error.log](https://github.com/user-attachments/files/15903277/browser-console-error.log)

- When using the TAB key to navigate if I close the onscreen keyboard that's treated as an incorrect answer:

https://github.com/learningequality/kolibri/assets/79847249/5443a1aa-b3a2-45e7-b826-b6668ed2c20d

rtibbles commented:
1 is definitely not related to the changes here, I have filed a follow up issue for it here: https://github.com/learningequality/morango/issues/224

rtibbles committed:
Don't attempt to submit an answer when the keypad is dismissed with the enter key.

rtibbles commented:
@pcenov I have fixed the inadvertent submission error you saw when dismissing the keypad, I was unable to replicate the other issue, and have filed a follow up here: https://github.com/learningequality/kolibri/issues/12322 if you could post the database from there so I can replicate, that would be helpful.

The error that is occurring is due to the answer not matching to the question, so I am trying to work out how that has happened.

marcellamaki commented:
re-testing of the mobile keyboard close input confirms that issue Peter noted is resolved

marcellamaki reviewed (approved):
Thank you, @rtibbles - with my last manual QA test, I've confirmed the remaining bug that Peter found is resolved
